test/specs/core/mergeConfig.spec.js

```diff
@@ -33,30 +33,35 @@ describe('core::mergeConfig', function() {
   it('should not inherit request options', function() {
     var localDefaults = {
       method: '__sample method__',
-      params: '__sample params__',
       data: { foo: true }
     };
     var merged = mergeConfig(localDefaults, {});
     expect(merged.method).toEqual(undefined);
-    expect(merged.params).toEqual(undefined);
     expect(merged.data).toEqual(undefined);
   });
 
-  it('should merge auth, headers, proxy with defaults', function() {
-    expect(mergeConfig({ auth: undefined }, { auth: { user: 'foo', pass: 'test' } })).toEqual({
+  it('should merge auth, headers, params, proxy with defaults', function() {
+    expect(mergeConfig({ auth: { user: 'foo' } }, { auth: { pass: 'test' } })).toEqual({
```

Comment:
@textbook Thanks for your quick update. And how about restoring the same for this case? Sorry I didn't mention it yesterday and thought that you could notice. :P

Reply:
That change was deliberate, previously it didn't actually test that they got *merged*.

Reply:
I think they have their meanings. The first case simulates the common usage, which means nothing in the defaults and users set a new config. The second case simulates the field `pass` is replaced with the new value.

```
    expect(mergeConfig({ auth: undefined }, { auth: { user: 'foo', pass: 'test' } })).toEqual({
      auth: { user: 'foo', pass: 'test' }
    });
    expect(mergeConfig({ auth: { user: 'foo', pass: 'test' } }, { auth: { pass: 'foobar' } })).toEqual({
      auth: { user: 'foo', pass: 'foobar' }
    });
```

Reply:
Ah, I see; that feels like a distinctive case to me, so I've split them apart into two tests. I also moved the loop over the merged params outside the test cases, to further reduce duplication.